### PR TITLE
Lazily import some dependencies

### PR DESCRIPTION
**Pull request by sk1p**
Mostly this prevents importing pytorch, matplotlib, skimage, sklearn at
import time. Local/lazy imports should still perform well (some ns), as they
reduce to something like a dict lookup.

This improves initial import time, measured as

    python -c 'import libertem.api'

by a factor of two. Running some of our slowest tests (tests/executor
and tests/server) is reduced from 264s to about 220s on my workstation. Let's see how the
CI time is improved, hopefully there the impact is even larger.

Something like the following was used to profile import times:

```
PYTHONPROFILEIMPORTTIME=1 python -c 'import libertem.api' 2>&1 | sort -k 3 -n
```

@uellue thoughts?

## TODO

- [ ] measure impact of `import torch` in `ApplyMasksUDF` - does it take time in the profile? Maybe move to `get_task_data`?

## Contributor Checklist:

* [ ] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases
* [ ] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if
changes were made to the GUI)

### CODE REVIEW

**Comment by uellue** [comment]
Nice! Reducing the ever-growing startup time is definitely a good idea. :-) Good to go from my perspective!

**Comment by sk1p** [comment]
Comparing https://travis-ci.com/github/LiberTEM/LiberTEM/builds/198721103 and https://travis-ci.com/github/LiberTEM/LiberTEM/builds/198751228, it seems to shave off 1m to 1m30s from our Linux builds, 3-5 mins from the  Mac OS builds (don't know how much variance there usually is...)

File: src/libertem/udf/masks.py [hunk 1]
```diff
@@ -144,10 +144,7 @@ def process_tile(self, tile):
         ''
         flat_data = tile.reshape((tile.shape[0], -1))
         if self.task_data.use_torch:
-            try:
```
Comment (uellue) [inline src/libertem/udf/masks.py]: The purpose of `try ... except` is to work without installed torch since that is an optional dependency.

Comment (uellue) [reply src/libertem/udf/masks.py]: Is that still required if use_torch is set, or is use_torch only set if torch can be imported?

Comment (sk1p) [reply src/libertem/udf/masks.py]: True, but when we enter that branch, `use_torch` is `True`, which is only the case if the import in `get_task_data` worked
